ceem/ceem.py

```diff
@@ -118,6 +118,19 @@ def step(self, xs, sys, smooth_solver_kwargs=None, learner_criterion_kwargs=None
         if isinstance(learner_opt_kwargs, dict):
             learner_opt_kwargs = [learner_opt_kwargs for _ in range(len(self._learning_criteria))]
 
+        if hp_schedulers is not None:
+            if isinstance(hp_schedulers, dict):
+                hp_schedulers = [hp_schedulers for _ in range(len(self._learning_criteria))]
+            else:
+                assert len(hp_schedulers) == len(self._learning_criteria)
+
+        if hp_schedulers is not None:
+            for i in range(len(self._learning_criteria)):
+                for hp, scheduler in hp_schedulers[i].items():
+                    newhp = scheduler.step()
```

Comment:
`step()` will be called multiple times in one epoch if line 123 is called right?

Reply:
Hmm. I should deepcopy it instead. Good catch!

Reply:
But the setup should happen in train, you are right.

Reply:
For the same criteria there can be multiple schedulers for the different hyper parameters, so this is fine?

Reply:
The issue I see is:
1. if you deep copy, then when you `step`, the original object from which you deepcopied won't stepped. 
2. if you have multiple criteria (eg dynamics, observations) but want to use the same scheduler for them, then (if you didn't deep copy) you would call `step` twice.

I filed a PR fix to this, maybe see if that works?

Reply:
I see what you did. Theres just one issue with the way you've set up `LambdaScheduler` I think. I'll point it out.